src/components/profile/EnvironmentForm.jsx

```diff
@@ -0,0 +1,218 @@
+import React, { Component } from "react";
+import PropTypes from "prop-types";
+import { withStyles } from "@material-ui/core";
+import { compose } from 'redux';
+import { connect } from "react-redux";
+import { getEnvironments } from "../../actions/environments";
+import DialogTitle from '@material-ui/core/DialogTitle';
+import Dialog from '@material-ui/core/Dialog';
+import TextField from '@material-ui/core/TextField';
+import Button from '@material-ui/core/Button';
+import api from '../../utils/api';
+import AddIcon from '@material-ui/icons/Add';
+import IconButton from '@material-ui/core/IconButton';
+import Input from '@material-ui/core/Input';
+import InputLabel from '@material-ui/core/InputLabel';
+import InputAdornment from '@material-ui/core/InputAdornment';
+import FormControl from '@material-ui/core/FormControl';
+import Chip from '@material-ui/core/Chip';
+import List from '@material-ui/core/List';
+import Typography from "@material-ui/core/Typography";
+
+
+const styles = theme => ({
+	root:{
+		flex: 1,
+	},
+	container: {
+		display: 'flex',
+		flexWrap: 'wrap',
+	},
+	textField: {
+		margin: theme.spacing.unit*2,
+	},
+	tagStyle: {
+		margin: theme.spacing.unit,
+	},
+})
+
+class EnvironmentForm extends Component {
+  constructor(props) {
+		super(props);
+		this.state = {
+			id: this.props.envExists ? this.props.environment.id : this.props.repository.id,
```

Comment:
you don't have to do `this.props` here, you can just use `props` since it's passed into the constructor.

Reply:
removed it here and elsewhere